### ISSUE TITLE
impl Display instead of ToString

### ISSUE THREAD

**Issue by andrewbanchich** [rust-lang/rust-clippy#12076]
### What it does

[The docs for `ToString`](https://doc.rust-lang.org/stable/alloc/string/trait.ToString.html) say:

> ToString shouldn’t be implemented directly: [Display](https://doc.rust-lang.org/stable/alloc/fmt/trait.Display.html) should be implemented instead

Currently there are no warnings to a direct `ToString` impl.

### Example

```rust
impl ToString for Point {
    fn to_string(&self) -> String {
        format!("({}, {})", self.x, self.y)
    }
}
```

Could be written as:

```rust
impl fmt::Display for Point {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "({}, {})", self.x, self.y)
    }
}
```

**Comment by andrewbanchich** [comment]
If we're open to the idea of adding this, I'd like to take this on.

**Comment by PartiallyTyped** [comment]
Feel free to use `@rustbot claim` :D

**Comment by andrewbanchich** [comment]
@rustbot claim

Thanks @PartiallyTyped !

**Comment by asquared31415** [comment]
This really seems like something that could be put in rustc directly, since the docs say to not implement it.  But it's much easier to get a lint into clippy than rustc and it can always be moved up to rustc.
